### DISCUSSION TITLE
How to add client?

### DISCUSSION THREAD

**Discussion by PrzemekSkw** [keycloak/keycloak#27006] [Q&A]
Hello, I read docs but don't understand how to add client (paperless-ngx). I want to add 2FA for that app. I install keycloak on unraid server as docker. I add it to Nginx Proxy Manager in https://keycloack.duckdns.net.

![keycloak_screen](https://github.com/keycloak/keycloak/assets/25219988/52627b91-3cee-4f52-82e3-915e74dcf0aa)
![Zrzut ekranu z 2024-02-14 14-30-25](https://github.com/keycloak/keycloak/assets/25219988/c51aa50d-aed6-4530-b02d-13920017061f)
![Zrzut ekranu z 2024-02-14 14-32-07](https://github.com/keycloak/keycloak/assets/25219988/2cc6ab82-2446-453c-b094-e7944d9f23a6)

I create realm, then user and then client but don't understand how to fill that form.
Don't know how to disable login credentials for paperless and how to fix redirect URL error. How to set that address fields.
Can somebody help me with that?
Regards.

**Comment by darius-m** [comment]
It seems that the server sends the request to Keycloak correctly, but Keycloak does not recognize the URI it's supposed to return the authentication code to. Usually the redirect URI looks like `https://example.com/oidc/redirect` (some path that is part of the authentication plugin that can use the authentication code).

Look up what the path should be for the client you are using, and set the full URL in the valid redirect URI field. The parameter is also included in the authentication request from your client as `redirect_uri` so you should be able to easily get it by looking at the request's URL.

**Reply by PrzemekSkw** [reply]
Hello, thank You for answer. I make some changes and now my situation looks like that:
1. I go to my paperless-ngx serwer on `https://paperless.....duckdns.org` and I see paperless login screen with keycloak button:
![Zrzut ekranu z 2024-02-14 14-30-25](https://github.com/keycloak/keycloak/assets/25219988/c58e220c-91cd-40c0-8bd3-26667efd14f6)
 
 I can login normal as paperless user or I can use keycloak login with user and TOTP code.
 I don't know how to make in keycloak settings to redirect my `https://paperless....duckdns.org` to my keycloak login.

**Reply by darius-m** [reply]
Keycloak should redirect you back to the client after receiving and validating an authentication request. Otherwise, Keycloak is not really involved. The paperless service must be configured to redirect you to Keycloak when pressing the button.

**Reply by PrzemekSkw** [reply]
Yes, that works ok but when I first time put my paperless address I get paperless login screen.
I want to have keylock login screen first. Not only button.
